src/battlegoals/battleGoalCard.tsx

```diff
@@ -29,14 +29,85 @@ export default class BattleGoalCard extends React.Component<BattleGoalCardProps,
   }
 
   public render(): React.ReactNode {
-    const style = {
+    const battleGoalStyle = {
+      position: 'relative',
       height: 300,
       width: 200
     } as React.CSSProperties;
-    const imageName = this.state.hidden ? battleGoalBackground : battleGoalByLocalId(this.props.battleGoalId).name;
-    const imageUrl = "https://raw.githubusercontent.com/any2cards/gloomhaven/master/images/battle-goals/" + imageName + ".png";
-    return <span onMouseOver={this.reveal} onMouseOut={this.hide}>
-                <img src={imageUrl} style={style}/>
-    </span>;
+
+    const imageStyle = {
+      borderRadius: '15px',
+      height: '100%',
+      width: '100%'
+    };
+
+    const imageUrl = (this.state.hidden && !this.props.show) ? battleGoalImages.background : battleGoalImages.foreground;
+    return <div style={battleGoalStyle} onMouseOver={this.reveal} onMouseOut={this.hide}>
+      <img style={imageStyle} src={imageUrl} alt='ups'/>
+      {this.renderOverlay()}
+    </div>;
+  }
+
+  private renderOverlay(): React.ReactNode {
+    if (this.state.hidden && !this.props.show) {
+      return null;
+    }
+
+    const overlayStyle = {
```

Comment:
I think all of these should be moved to a .css file, unless you're dynamically changing css attribute (which it seems you're not)

Reply:
Where do you see the benefit having the styles in two different places?

Reply:
Hm I think the code "logic" should deal with layout / ui / styling as few as possible. This way, the code is more concise and readable and "in one place". 

I thought this project already had a CSS file but I see that is not the case. (apart from the very minimalistic one you just added). So I guess it's consistent to put the styles inline for now. (Can be a follow-up item if anyone wants to change this)